commit 7b4481818fe920904457496db40d59dd3d0a83bb
Author: Matthias Reichl <hias@horus.com>
Date:   Sun Oct 11 15:35:20 2015 +0200

    bcm2835-i2s: Register PCM device
    
    Code ported from bcm2708-i2s driver in Raspberry Pi tree.
    
    RPi commit ba46b4935a23aa2caac1855ead52a035d4776680 ("ASoC: Add
    support for BCM2708")
    
    This driver adds support for digital audio (I2S)
    for the BCM2708 SoC that is used by the
    Raspberry Pi. External audio codecs can be
    connected to the Raspberry Pi via P5 header.
    
    It relies on cyclic DMA engine support for BCM2708.
    
    Signed-off-by: Florian Meier <florian.meier@koalo.de>
    
    Signed-off-by: Matthias Reichl <hias@horus.com>

diff --git a/sound/soc/bcm/bcm2835-i2s.c b/sound/soc/bcm/bcm2835-i2s.c
index aab3df9..0e5c787 100644
--- a/sound/soc/bcm/bcm2835-i2s.c
+++ b/sound/soc/bcm/bcm2835-i2s.c
@@ -799,6 +799,25 @@ static const struct snd_soc_component_driver bcm2835_i2s_component = {
 	.name		= "bcm2835-i2s-comp",
 };
 
+static const struct snd_pcm_hardware bcm2835_pcm_hardware = {
+	.info			= SNDRV_PCM_INFO_INTERLEAVED |
+				  SNDRV_PCM_INFO_JOINT_DUPLEX,
+	.formats		= SNDRV_PCM_FMTBIT_S16_LE |
+				  SNDRV_PCM_FMTBIT_S24_LE |
+				  SNDRV_PCM_FMTBIT_S32_LE,
+	.period_bytes_min	= 32,
+	.period_bytes_max	= 64 * PAGE_SIZE,
+	.periods_min		= 2,
+	.periods_max		= 255,
+	.buffer_bytes_max	= 128 * PAGE_SIZE,
+};
+
+static const struct snd_dmaengine_pcm_config bcm2835_dmaengine_pcm_config = {
+	.prepare_slave_config = snd_dmaengine_pcm_prepare_slave_config,
+	.pcm_hardware = &bcm2835_pcm_hardware,
+	.prealloc_buffer_size = 256 * PAGE_SIZE,
+};
+
 static int bcm2835_i2s_probe(struct platform_device *pdev)
 {
 	struct bcm2835_i2s_dev *dev;
@@ -870,7 +889,9 @@ static int bcm2835_i2s_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	ret = devm_snd_dmaengine_pcm_register(&pdev->dev, NULL, 0);
+	ret = devm_snd_dmaengine_pcm_register(&pdev->dev,
+			&bcm2835_dmaengine_pcm_config,
+			SND_DMAENGINE_PCM_FLAG_COMPAT);
 	if (ret) {
 		dev_err(&pdev->dev, "Could not register PCM: %d\n", ret);
 		return ret;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1017_7b4481818fe920904457496db40d59dd3d0a83bb.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1017_7b4481818fe920904457496db40d59dd3d0a83bb.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
